great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3917,27 +3917,28 @@ def get_available_data_asset_names(
             ):  # Iterate over both together
                 for idx, datasource_name in enumerate(datasource_names):
                     datasource = self.get_datasource(datasource_name)
-                    assert not isinstance(
```

Comment:
@jshaikGX Before we can confirm this approach, we need to make sure that we are not breaking anything.  The problem is that the existing implementation is based on GX V2!  So the question is -- will anything break that we are still supposed to support?  Perhaps something needs to be done to ensure that whoever is relying on the V2 construct `batch_kwargs_generator_names` (or anything to do with `batch_kwargs_generator`) should not be.  Maybe some kind of a warning or a deprecation notice is in order?  We never supported this call `get_available_data_asset_names()` at the DataContext level properly in V3 (only in Datasource) which is what you are using, but this will leave the V2 implementation non-existent.  Recently, @anthonyburdi deleted quite a bit of V2 functionality, please add him as a reviewer.  Thank you.

Reply:
@anthonyburdi

Reply:
I haven't yet looked at the code in this PR, but just to respond to this comment - yes we have deprecated the V2 api several minor versions ago and have started to delete some of the code per our deprecation policy (with the rest being deleted hopefully soon).

Reply:
Also, I would suggest adding the `deprecated_argument` decorator to this method for `batch_kwargs_generator_names`, looks like we missed adding that when adding this method to the public API.

Reply:
Also, this does not look like it will break existing v2 api workflows yet, there are isinstance checks for `FluentDatasource`.

Reply:
yeah this shouldn't break V2 - that being said, I don't want to add scope creep to @jshaikGX 's work here.

Are we confident in this deprecation? Should we surface in arch to confirm?